capi/cbindgen.toml

```diff
@@ -4,6 +4,29 @@ language = "C"
 cpp_compat = true
 include_guard = "__blazesym_h_"
 usize_is_size_t = true
+after_includes = """\
+/* Helper macro to declare and initialize a blazesym input struct
+ *
+ * This dance with uninitialized declaration, followed by memset to zero,
+ * followed by assignment using compound literal syntax is done to preserve
+ * ability to use a nice struct field initialization syntax and **hopefully**
+ * have all the padding bytes initialized to zero. It's not guaranteed though,
+ * when copying literal, that compiler won't copy garbage in literal's padding
+ * bytes, but that's the best way I've found and it seems to work in practice.
+ *
+ * Macro declares opts struct of given type and name, zero-initializes,
+ * including any extra padding, it with memset() and then assigns initial
+ * values provided by users in struct initializer-syntax as varargs.
+ */
+#define BLAZE_INPUT(TYPE, NAME, ...)        \\
+  struct TYPE NAME = ({                     \\
+    memset(&NAME, 0, sizeof(struct TYPE));  \\
+    (struct TYPE) {                         \\
+      ._size = sizeof(struct TYPE),         \\
```

Comment:
very subjective, but `_size` doesn't look great due to underscore, tbh. I suspect `size` is painful to use from Rust side, right? Do you hate `sz`?

Reply:
I wouldn't say it's painful to use, it just overlaps with the `size` member of `blaze_sym_info`. But yeah, not a fan of `sz`. If anything it makes it more confusing from my point of view, because while the underscore is kind of sort of meant to indicate irrelevance to the user (they use a macro), `sz` is just an abbreviated `size`.

I suppose we could rename `blaze_sym_info.size` to `blaze_sym_info.sym_size` and then go with `size`, but not sure if that fundamentally changes much for the better. Or we could call `_size` `type_size` or something like that.

Reply:
out of those options, I'd probably vote for renaming to sym_size (it makes sense), and use cleaner "size" field name. That underscore is just an eye sore, tbh.

Reply:
I went with the `type_size` approach.